x/amm/keeper/initialize_pool.go

```diff
@@ -0,0 +1,54 @@
+package keeper
+
+import (
+	"fmt"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	banktypes "github.com/cosmos/cosmos-sdk/x/bank/types"
+	"github.com/elys-network/elys/x/amm/types"
+)
+
+// This function:
+// - saves the pool to state
+// - Mints LP shares to the pool creator
+// - Sets bank metadata for the LP denom
+// - Records total liquidity increase
+// - Calls the AfterPoolCreated hook
+func (k Keeper) InitializePool(ctx sdk.Context, pool *types.Pool, sender sdk.AccAddress) (err error) {
+	// Mint the initial pool shares share token to the sender
+	err = k.MintPoolShareToAccount(ctx, pool, sender, pool.GetTotalShares().Amount)
+	if err != nil {
+		return err
+	}
+
+	// Finally, add the share token's meta data to the bank keeper.
+	poolShareBaseDenom := types.GetPoolShareDenom(pool.GetPoolId())
```

Comment:
What is the different between poolsharebase denom and pool display denom. pool share = lp token denom?